File: Portfolio/3.1P - Clock Class📕/Clock/Clock/Counter.cs
using System;

namespace Clock
{
    public class Counter
    {
        private int _count;
        private string _name;

        public Counter(string name)
        {
            _count = 0;
            _name = name;
        }

        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }

        public string Name
        {
            get 
            {
                return _name;
            }
            set 
            {
                _name = value;
            }
        }

        public int Tick
        {
            get 
            {
                return _count;
            }
        }
    }
}

File: Portfolio/11.1P - Clock in Another Language📕/Clock/Counter.rb
module Clock
    public
    class Counter
        attr_accessor :name, :tick

        public
        def initialize(name)
            @tick = 0
            @name = name
        end

        public
        def increment()
            @tick += 1
        end

        public
        def reset()
            @tick = 0
        end
    end
end